src/_pytest/python.py

```diff
@@ -1419,16 +1414,6 @@ def __init__(
 
     def _initrequest(self):
         self.funcargs = {}
-        if self._isyieldedfunction():
-            assert not hasattr(
-                self, "callspec"
-            ), "yielded functions (deprecated) cannot have funcargs"
-        else:
-            if hasattr(self, "callspec"):
-                callspec = self.callspec
-                assert not callspec.funcargs
-                if hasattr(callspec, "param"):
-                    self.param = callspec.param
```

Comment:
This no longer seems to be used anywhere, the `param` attribute would be added by `metafunc.addcall`, which no longer exists:

https://github.com/pytest-dev/pytest/blob/91b6f2bda8668e0f74190ed223a9eec01d09a0a7/_pytest/python.py#L888-L891